### DISCUSSION TITLE
dmidecode for Flatcar Linux

### DISCUSSION THREAD

**Discussion by ahjohannessen** [flatcar/Flatcar#1282] [General]
I wonder if [dmidecode](https://www.nongnu.org/dmidecode/) is available for Flatcar Linux - Reason for asking is that I use [Nomad](https://www.nomadproject.io) scheduler/orchestrator on Flatcar machines and learned that Nomad v1.7.0+ requires dmidecode  to probe systems about resources like cpu / memory and so on.

**Comment by jepio** [comment]
It's not available yet, but could be added. I've opened https://github.com/flatcar/Flatcar/issues/1285 to track this

**Reply by ArKam** [reply]
That could be a sysext image tho.

**Comment by olljanat** [comment]
@ahjohannessen how you deploy Nomad to Flatcar?

I’m thinking if would make sense to create sysext image which would provide whole HashiStack (Nomad + Consul + Vault).

Especially now when [flatcar/nomad-on-flatcar](https://github.com/flatcar-archive/nomad-on-flatcar) is archived this probably should be included to [flatcar/sysext-bakery](https://github.com/flatcar/sysext-bakery)

dmidecode and other mandatory/useful tools of course can be included to same image.

**Comment by ahjohannessen** [comment]
I install it using ansible-nomad - However, most of my machines are now Fedora CoreOS / uCore as rpm-ostree allows me to install python / unzip / etc. + use zfs for select machines (via uCore).

**Reply by olljanat** [reply]
Thanks for quick answer. Looks that we are going opposite directions because I'm looking for if we can rid of Ansible by migrating to Flatcar 😃

**Reply by ahjohannessen** [reply]
I try to keep things as minimal as possible, but ansible is to be required for some things in our setup, so I do this as a baseline:

`rpm-ostree install --allow-inactive python unzip dmidecode ` in a one shot systemd unit.

for the zfs machines I do something like:

`/etc/coreos-zfs-autorebase/coreos-zfs-autorebase.sh` in another one shot systemd unit.


```shell
- path: /etc/coreos-zfs-autorebase/coreos-zfs-autorebase.sh
  contents:
    inline: |
      #!/usr/bin/bash
      echo "Rebasing to Fedora CoreOS with ZFS in 5 seconds"
      sleep 5
      rpm-ostree rebase --install=unzip --install=dmidecode ostree-unverified-registry:ghcr.io/ublue-os/fedora-coreos:stable-zfs \
        && touch /etc/coreos-zfs-autorebase/.complete \
        && systemctl disable coreos-zfs-autorebase.service \
        && echo zfs > /etc/modules-load.d/zfs.conf \
        && systemctl reboot
  mode: 0754
  ```
If I used something else than ansible that worked better with Flatcar I would not have big reasons to switch.
